alchemy/src/aws/control/resource.ts

```diff
@@ -0,0 +1,328 @@
+      const resources = await client.listResources(`AWS::${serviceName}::*`);
+
+      // Delete each resource and track operations
+      const deletionPromises = (
+        resources.resources as CloudControlResourceInfo[]
+      ).map(async (resource) => {
+        try {
+          console.log(
+            `Deleting resource ${resource.identifier} (${resource.typeName})`
+          );
+
+          const response = await client.deleteResource(
+            resource.typeName,
+            resource.identifier
+          );
+
+          // Wait for deletion to complete
+          const result = await response;
+
+          if (result.status === "FAILED") {
+            console.error(
+              `Failed to delete resource ${resource.identifier}: ${result.message}`
+            );
+          } else {
+            console.log(`Successfully deleted resource ${resource.identifier}`);
+          }
+        } catch (error: unknown) {
+          const cloudError = error as CloudControlError;
+          if (cloudError.code === "ResourceNotFoundException") {
+            // Resource already deleted, not an error
+            console.log(`Resource ${resource.identifier} already deleted`);
+          } else {
+            console.error(
+              `Error deleting resource ${resource.identifier}:`,
+              error
+            );
+          }
+        }
+      });
+
+      // Wait for all deletions to complete
+      await Promise.allSettled(deletionPromises);
+    } catch (error: unknown) {
+      const cloudError = error as CloudControlError;
+      if (cloudError.code === "ResourceNotFoundException") {
+        // No resources found, not an error
+        console.log(`No resources found matching pattern ${pattern}`);
+      } else {
+        console.error(`Error listing resources for pattern ${pattern}:`, error);
+      }
+    }
+  }
+);
+
+// Cache for memoizing resource handlers
+const resourceHandlers: Record<string, any> = {};
+
+/**
+ * Creates a memoized Resource handler for a CloudFormation resource type
+ *
+ * @param typeName CloudFormation resource type (e.g., "AWS::S3::Bucket")
+ * @returns A memoized Resource handler for the specified type
+ */
+export function createResourceType(typeName: string) {
+  // Return cached handler if it exists
+  if (resourceHandlers[typeName]) {
+    return resourceHandlers[typeName];
+  }
+
+  // Create a new handler
+  const handler = Resource(
+    typeName,
+    async function (this: Context<any>, id: string, props: any): Promise<any> {
+      // Initialize the Cloud Control API client
+      const client = createCloudControlClient();
+
+      if (this.phase === "delete") {
+        try {
+          if (this.output?.id) {
+            await client.deleteResource(typeName, this.output.id);
+          }
+        } catch (error) {
+          // Log but don't throw on deletion errors
+          console.error(`Error deleting resource ${typeName}:`, error);
```

Comment:
I think we only want to swallow errors that indicate the resource is already deleted. If we have a transient failure, deletion should fail and the user should retry.